patroni/api.py

```diff
@@ -110,6 +113,45 @@ def do_GET_patroni(self):
         response = self.get_postgresql_status(True)
         self._write_status_response(200, response)
 
+    def do_GET_config(self):
+        self._write_json_response(200, self.server.patroni.config.dynamic_configuration)
+
+    @staticmethod
+    def _patch_config(config, data):
+        for name, value in data.items():
+            if isinstance(value, dict) and name in config:
+                RestApiHandler._patch_config(config[name], value)
+            elif value is None:
+                config.pop(name, None)
+            else:
+                config[name] = value
+
+    @check_auth
+    def do_PATCH_config(self):
+        content_length = int(self.headers.get('content-length', 0))
+        request = json.loads(self.rfile.read(content_length).decode('utf-8'))
+        cluster = self.server.patroni.ha.dcs.get_cluster()
+        data = cluster.config.data.copy()
+        RestApiHandler._patch_config(data, request)
+        if data != cluster.config.data:
```

Comment:
>  and how are you going to compare int and string?
I assume that you won't get int and string together for a single parameter value, would you? Otherwise, I don't understand your question. You won't catch some more complicated cases, like changing '1MB' shared buffers to '1024KB', but this is not the goal: the goal is to avoid unnecessary restarts when the value being changed is accidentally quoted...